File: src/main/java/com/github/rabend/generators/ArrayGenerator.java
package com.github.rabend.generators;

import com.fasterxml.jackson.databind.JsonNode;

import java.util.HashSet;
import java.util.Set;
import java.util.concurrent.ThreadLocalRandom;
import java.util.stream.Collectors;

public class ArrayGenerator extends AbstractValueGenerator {
    @Override
    public String generateRandomValue(final JsonNode node) {
        JsonNode itemsNode = node.get("items");
        AbstractValueGenerator itemGenerator = ValueGeneratorsLookup.getGeneratorForType(itemsNode.get("type").asText());

        int arrayCount = ThreadLocalRandom.current().nextInt(6);

        if (itemsNode.has("enum")) {
            arrayCount = itemsNode.withArray("enum").size();
        }

        Set<String> values = new HashSet<>();
        for (int i = 0; i < arrayCount; i++) {
            values.add(itemGenerator.generateRandomValue(itemsNode));
        }

        return values.stream().collect(Collectors.joining(",", "[", "]"));
    }
}

File: src/main/java/com/github/rabend/generators/ArrayGenerator.kt
package com.github.rabend.generators

import com.fasterxml.jackson.databind.JsonNode
import java.util.concurrent.ThreadLocalRandom
import java.util.stream.Collectors

class ArrayGenerator : AbstractValueGenerator() {
    override fun generateRandomValue(node: JsonNode?): String? {
        val itemsNode = node!!["items"]
        val itemGenerator = ValueGeneratorsLookup.getGeneratorForType(itemsNode["type"].asText())
        var arrayCount = ThreadLocalRandom.current().nextInt(6)
        if (itemsNode.has("enum")) {
            arrayCount = itemsNode.withArray<JsonNode>("enum").size()
        }
        val values: MutableSet<String?> = HashSet()
        for (i in 0 until arrayCount) {
            values.add(itemGenerator.generateRandomValue(itemsNode))
        }
        return values.stream().collect(Collectors.joining(",", "[", "]"))
    }
}
